org.eclipse.xtext/src/org/eclipse/xtext/resource/containers/LiveShadowedChunkedContainer.java

```diff
@@ -0,0 +1,151 @@
+				|| (getProjectConfig() != null && getProjectConfig().findSourceFolderContaining(uri) != null);
+	}
+
+	@Override
+	public IResourceDescription getResourceDescription(URI uri) {
+		return isContained(uri) ? descriptions.getResourceDescription(uri) : null;
+	}
+
+	@Override
+	public int getResourceDescriptionCount() {
+		return size(getResourceDescriptions());
+	}
+
+	@Override
+	public Iterable<IResourceDescription> getResourceDescriptions() {
+		Set<URI> localURIs = getExistingOrRenamedResourceURIs();
+		return Iterables.concat(getContainedLocalDescriptions(),
+				filter(getChunk().getAllResourceDescriptions(), it -> !localURIs.contains(it.getURI())));
+	}
+
+	@Override
+	public boolean hasResourceDescription(URI uri) {
+		return getChunk().getResourceDescription(uri) != null;
+	}
+
+	@Override
+	public Iterable<IEObjectDescription> getExportedObjects() {
+		Set<URI> localURIs = getExistingOrRenamedResourceURIs();
+		return Iterables.concat(Iterables.concat(map(getContainedLocalDescriptions(), it -> it.getExportedObjects())),
+				filter(getChunk().getExportedObjects(), it -> !localURIs.contains(it.getEObjectURI().trimFragment())));
+	}
+
+	@Override
+	public Iterable<IEObjectDescription> getExportedObjects(EClass type, QualifiedName name, boolean ignoreCase) {
+		Set<URI> localURIs = getExistingOrRenamedResourceURIs();
+		return Iterables.concat(
+				Iterables.concat(
+						map(getContainedLocalDescriptions(), it -> it.getExportedObjects(type, name, ignoreCase))),
+				filter(getChunk().getExportedObjects(type, name, ignoreCase),
+						it -> !localURIs.contains(it.getEObjectURI().trimFragment())));
+	}
+
+	@Override
+	public Iterable<IEObjectDescription> getExportedObjectsByObject(final EObject object) {
+		Set<URI> localURIs = getExistingOrRenamedResourceURIs();
+		return Iterables.concat(
+				Iterables.concat(map(getContainedLocalDescriptions(), it -> it.getExportedObjectsByObject(object))),
+				filter(getChunk().getExportedObjectsByObject(object),
+						it -> !localURIs.contains(it.getEObjectURI().trimFragment())));
+	}
+
+	@Override
+	public Iterable<IEObjectDescription> getExportedObjectsByType(EClass type) {
+		Set<URI> localURIs = getExistingOrRenamedResourceURIs();
+		return Iterables.concat(
+				Iterables.concat(map(getContainedLocalDescriptions(), it -> it.getExportedObjectsByType(type))),
+				filter(getChunk().getExportedObjectsByType(type),
+						it -> !localURIs.contains(it.getEObjectURI().trimFragment())));
+	}
+
+	@Override
+	public boolean isEmpty() {
+		return IterableExtensions.isEmpty(getContainedLocalDescriptions()) && getChunk().isEmpty();
+	}
+
+	protected Set<URI> getExistingOrRenamedResourceURIs() {
+		ResourceSet resourceSet = descriptions.getResourceSet();
+		if (resourceSet instanceof ResourceSetImpl)
+			return ((ResourceSetImpl) resourceSet).getURIResourceMap().keySet();
+		throw new IllegalStateException("ResourceSet is not a ResourceSetImpl");
```

Comment:
"ResourceSet is not a ResourceSetImpl" -> "ResourceSet is not an instance of ResourceSetImpl"

Reply:
Could you please be more specific!? I am not sure what I should do with this comment.

Reply:
I was suggesting to replace the detailed message with - ResourceSet is not an instance of ResourceSetImpl